onmt/bin/train.py

```diff
@@ -53,6 +52,33 @@ def prepare_fields_transforms(opt):
     return fields, transforms_cls
 
 
+def prepare_update_fields_transforms(opt, old_fields):
+    """Prepare or dump fields & transforms before training."""
+    transforms_cls = get_transforms_cls(opt._all_transform)
+    specials = get_specials(opt, transforms_cls)
+    fields = build_update_dynamic_fields(
+        opt, old_fields, src_specials=specials['src'], tgt_specials=specials['tgt'])
+
+    # maybe prepare pretrained embeddings, if any
+    prepare_pretrained_embeddings(opt, fields)
+
+    if opt.dump_fields:
+        save_fields(fields, opt.save_data, overwrite=opt.overwrite)
+    if opt.dump_transforms or opt.n_sample != 0:
+        transforms = make_transforms(opt, transforms_cls, fields)
+    if opt.dump_transforms:
+        save_transforms(transforms, opt.save_data, overwrite=opt.overwrite)
+    if opt.n_sample != 0:
+        logger.warning(
+            "`-n_sample` != 0: Training will not be started. "
+            f"Stop after saving {opt.n_sample} samples/corpus.")
+        save_transformed_sample(opt, transforms, n_sample=opt.n_sample)
+        logger.info(
+            "Sample saved, please check it before restart training.")
+        sys.exit()
```

Comment:
Why duplicating this?

Reply:
It is a copy of `prepare_fields_transforms` which is only run when not checkpoint is given (fresh start). So I saw the need to give those options when updating vocabulary too. Maybe it could be generalized?